core/src/main/scala/kafka/log/LogConfig.scala

```diff
@@ -115,67 +117,214 @@ object LogConfig {
+        val b = new StringBuilder
+        b.append("<table class=\"data-table\"><tbody>\n")
+        b.append("<tr>\n")
+        b.append("<th>Name</th>\n")
+        b.append("<th>Description</th>\n")
+        b.append("<th>Type</th>\n")
+        b.append("<th>Default</th>\n")
+        b.append("<th>Valid Values</th>\n")
+        b.append("<th>Server Default Property</th>\n")
+        b.append("<th>Importance</th>\n");
+        b.append("</tr>\n");
+        for (config <- configs.asScala.toList) {
+            b.append("<tr>\n")
+            b.append("<td>")
+            b.append(config.name)
+            b.append("</td>")
+            b.append("<td>")
+            b.append(config.documentation)
+            b.append("</td>")
+            b.append("<td>")
+            b.append(config.`type`.toString().toLowerCase(Locale.ROOT))
+            b.append("</td>")
+            b.append("<td>")
+            if (config.hasDefault()) {
+                if (config.defaultValue == null)
+                    b.append("null")
+                else if (config.`type` == ConfigDef.Type.STRING && config.defaultValue.toString().isEmpty())
+                    b.append("\"\"")
+                else
+                    b.append(config.defaultValue)
+            } else
+                b.append("")
+            b.append("</td>")
+            b.append("<td>")
+            b.append(if (config.validator != null) config.validator.toString() else "")
+            b.append("</td>")
+            b.append("<td>")
+            b.append(serverDefaultConfigNames.get(config.name).get)
+            b.append("</td>")
+            b.append("<td>")
+            b.append(config.importance.toString().toLowerCase(Locale.ROOT))
+            b.append("</td>")
+            b.append("</tr>\n")
+        }
+        b.append("</tbody></table>")
+        b.toString
+    }
+
+  }
 
-  private val configDef = {
+  private val configDef: LogConfigDef = {
     import org.apache.kafka.common.config.ConfigDef.Importance._
     import org.apache.kafka.common.config.ConfigDef.Range._
     import org.apache.kafka.common.config.ConfigDef.Type._
     import org.apache.kafka.common.config.ConfigDef.ValidString._
 
-    new ConfigDef()
-      .define(SegmentBytesProp, INT, Defaults.SegmentSize, atLeast(Message.MinMessageOverhead), MEDIUM, SegmentSizeDoc)
-      .define(SegmentMsProp, LONG, Defaults.SegmentMs, atLeast(0), MEDIUM, SegmentMsDoc)
-      .define(SegmentJitterMsProp, LONG, Defaults.SegmentJitterMs, atLeast(0), MEDIUM, SegmentJitterMsDoc)
-      .define(SegmentIndexBytesProp, INT, Defaults.MaxIndexSize, atLeast(0), MEDIUM, MaxIndexSizeDoc)
-      .define(FlushMessagesProp, LONG, Defaults.FlushInterval, atLeast(0), MEDIUM, FlushIntervalDoc)
-      .define(FlushMsProp, LONG, Defaults.FlushMs, atLeast(0), MEDIUM, FlushMsDoc)
+    new LogConfigDef()
+      .define(SegmentBytesProp, INT, Int.box(Defaults.SegmentSize), atLeast(Message.MinMessageOverhead), MEDIUM,
```

Comment:
@wushujames, I don't think you need it once you make the `Object` -> `Any` change I suggested in the `define` method.